libGDX/core/src/com/mygdx/game/movement/pieces/Piece.java

```diff
@@ -13,7 +14,7 @@ public interface Piece {
   /**
    * @return true if your piece. false if enemy piece
    */
-  boolean isPlayerPiece();
+  boolean isPlayerPiece(Player p);
```

Comment:
Adding a parameter here doesn't seem necessary.

The Pieces will take in a boolean/Player on their construction so they can return the correct value here.

To simplify things, the enemy won't have their own 'Player' object so there will also only be **one** Player, if this returns true it is their Piece, otherwise its part of the enemy faction.

Reply:
The enemy can move pieces too.  The enemy has mana too. The enemy can place summoned pieces. It seems the enemy is a player.  Otherwise, if this is confusing, we should consider renaming it.